### ISSUE TITLE
[Beta 2] query keyword is required in all queries

### ISSUE THREAD

**Issue by Kyle-Mendes** [zino-app/graphql-flutter#235]
**Describe the bug**
Previously, you could make a query like so:
```dart
const String getUsers = """
  {
    users {
      id
      name
    }
  }
"""
```

however, now the `query` keyword is required, even if no variables are being passed.

```dart
const String getUsers = """
  query getUsers {
    users {
      id
      name
    }
  }
"""
```

**To Reproduce**
1. Attempt to make the first query (any query without `query`
2. *Failure*

**Expected behavior**
`query` should not be required.

**Comment by wizcas** [comment]
+1. Query builder should support default behavior of no `query` keyword.

Moreover, request would fail by unknown operation if there was a parameter list after `query` keyword without query name. 

For example this query **doesn't work**:
``` graphql
query ($order_by_time: Boolean = true) {
    items(order_by_time: $order_by_time){
        id
        name
    }
}
```

It has to be:
``` graphql
query GetItemsByOrder ($order_by_time: Boolean = true) {
    items(order_by_time: $order_by_time){
        id
        name
    }
}
```

**Comment by patelmanjeet** [comment]
When we use query or mutation without Operation name it will add operation name like 
`_operationName ??= 'UNNAMED/' + document.hashCode.toString();`

And my backend hasura graphql server gives error in **invalid operation**

**Comment by endigo** [comment]
It's not in just query when I run mutation from bloc (not using mutation widget) throws this error.

**Comment by HofmannZ** [comment]
@Kyle-MendeHow is this a valid query?

```dart
const String getUsers = """
  users {
    id
    name
  }
"""
```

Even if the `query` keyword is omitted shouldn't there always be wrapping brackets like so:

```dart
const String getUsers = """
  {
    users {
      id
      name
    }
  }
"""
```

**Comment by Kyle-Mendes** [comment]
@HofmannZ Oops! I accidentally typed it out wrong :) Good catch!

**Comment by micimize** [comment]
yeah this looks like a regression - probably need to decouple `operationName` from the operation identification functionality I added the default for.

I've been a bit busy, but I should be able to get to this later this week

**Comment by endigo** [comment]
@micimize any progress?

**Comment by micimize** [comment]
@endigo I haven't had time to work on it - I would update here proactively if I had

**Comment by micimize** [comment]
closed by #276
